woocommerce/woocommerce
# Deprecate report hooks

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:




Deprecate hooks related to legacy Reports.

This is part of an effort to deprecate legacy reports from the code perspective and eventually retire the feature completely.
This PR is targeting a feature branch. We will gather all reviewed code there, to be released together once we know for sure which version we target and how we would guide 3PDs towards the new Analytics.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable `WP_DEBUG`
2. Add a snippet that uses any of the deprecated hooks
```php
function noop( $t ) { return $t; }
add_filter( 'woocommerce_reports_screen_ids', 'noop' );
add_filter( 'woocommerce_reports_order_statuses', 'noop' );

add_filter( 'woocommerce_reports_get_order_report_data_args', 'noop' );
add_filter( 'woocommerce_reports_get_order_report_query', 'noop' );
add_filter( 'woocommerce_reports_get_order_report_data', 'noop' );
add_filter( 'woocommerce_admin_user_actions', 'noop' );
add_filter( 'woocommerce_admin_report_customer_list_user_query_args', 'noop' );
add_filter( 'woocommerce_admin_report_customers_user_query_args', 'noop' );
add_filter( 'woocommerce_admin_downloads_report_downloads_per_page', 'noop' );
add_filter( 'woocommerce_report_downloads_query_from', 'noop' );
add_filter( 'woocommerce_report_low_in_stock_query_from', 'noop' );
add_filter( 'woocommerce_report_most_stocked_query_from', 'noop' );
add_filter( 'woocommerce_report_out_of_stock_query_from', 'noop' );
add_filter( 'woocommerce_report_sales_by_category_get_products_in_category', 'noop' );
add_filter( 'woocommerce_admin_report_data', 'noop' );
add_filter( 'woocommerce_admin_report_chart_data', 'noop' );
add_filter( 'woocommerce_admin_stock_report_product_actions', 'noop' );
add_filter( 'woocommerce_admin_stock_report_products_per_page', 'noop' );
add_filter( 'woocommerce_reports_taxes_tax_rate', 'noop' );
add_filter( 'woocommerce_reports_taxes_rate', 'noop' );
add_action( 'woocommerce_admin_user_actions_start', 'noop' );
add_action( 'woocommerce_admin_user_actions_end', 'noop' );
```
4. Check for deprecation warnings.
   - `woocommerce_reports_screen_ids` - any wp-admin page
   - `woocommerce_reports_order_statuses` - a page that has a chart, like `/wp-admin/admin.php?page=wc-reports&tab=orders`
   - `woocommerce_reports_get_order_report_data_args, woocommerce_reports_get_order_report_query, woocommerce_reports_get_order_report_data, woocommerce_admin_report_data, woocommerce_admin_report_chart_data` - orders report `/wp-admin/admin.php?page=wc-reports&tab=orders`
   - `woocommerce_admin_user_actions, woocommerce_admin_user_actions_start, woocommerce_admin_user_actions_end, woocommerce_admin_report_customer_list_user_query_args` - Customers list `/wp-admin/admin.php?page=wc-reports&tab=customers&report=customer_list`
   - `woocommerce_admin_report_customers_user_query_args` - Customers `/wp-admin/admin.php?page=wc-reports&tab=customers&report=customers`
   - `woocommerce_admin_downloads_report_downloads_per_page, woocommerce_report_downloads_query_from` - Customer downloads `/wp-admin/admin.php?page=wc-reports&tab=orders&report=downloads`
   - `woocommerce_report_most_stocked_query_from, woocommerce_admin_stock_report_products_per_page, woocommerce_admin_stock_report_product_actions` - Most stocked `/wp-admin/admin.php?page=wc-reports&tab=stock&report=most_stocked`
   - `woocommerce_report_low_in_stock_query_from` - Low in stock `/wp-admin/admin.php?page=wc-reports&tab=stock`
   - `woocommerce_report_out_of_stock_query_from` - Out of stock `/wp-admin/admin.php?page=wc-reports&tab=stock&report=out_of_stock`
   - `woocommerce_report_sales_by_category_get_products_in_category` - Sales by category, pick a category, click "Show" `/wp-admin/admin.php?page=wc-reports&tab=orders&report=sales_by_category`
   - `woocommerce_reports_taxes_tax_rate, woocommerce_reports_taxes_rate` - Taxes `/wp-admin/admin.php?page=wc-reports&tab=taxes&report=taxes_by_code&range=year` 
	 Checking Taxes is a bit tricky with HPOS enabled. The following snippet could help:
	   ```php
		add_filter( 'woocommerce_reports_order_statuses', function( $t ) {
		    return [];
		} );
		add_filter( 'woocommerce_reports_get_order_report_data_args', function( $args ) {
		    $args['order_types'][] = 'shop_order_placehold';
		    return $args;
		} );
		```




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
